src/components/viewPartsButton.jsx

```diff
@@ -1,10 +1,23 @@
 import { useState } from "react";
 import { Button, Modal, List } from "antd";
 import { DownloadOutlined } from "@ant-design/icons";
+import PropTypes from "prop-types";
 
-const ViewParts = () => {
+const ViewParts = ({ setFileFor3dModel }) => {
   const [modalOpen, setModalOpen] = useState(false);
   const [partsData, setPartsData] = useState([]);
+  const [fileNameForUpload, setFileNameForUpload] = useState("");
+
+  //not sure if this fetch call was the best approach
+  //It's the only way i could get the entire file at the moment
+  const uploadPartTo3dModel = (part_id) => {
+    fetch(`/api/v0/parts/${part_id}/file`)
+      .then((res) => res.blob())
+      .then((blob) => {
+        const file = new File([blob], fileNameForUpload, { type: blob.type });
+        setFileFor3dModel(file);
+      });
+  };
```

Comment:
@ErikAmerico after you make changes based on my comments, just make a reply here saying that you did it so that I know that I can 'Resolve Conversation' just so we can keep the PR a bit cleaner and close all the conversations we're done with

I'm pretty sure that you did this one so Im going to resolve it!